tabs-tabs-tabs/tabs.js

```diff
@@ -1,23 +1,39 @@
+function firstUnpinnedTab(tabs) {
+  for (var tab of tabs) {
+    if (!tab.pinned) {
+      return tab.index;
+    }
+  }
+}
+
 document.addEventListener("click", function(e) {
   function callOnActiveTab(callback) {
     chrome.tabs.query({}, function(tabs) {
       for (var tab of tabs) {
         if (tab.active) {
-          callback(tab);
+          callback(tab, tabs);
```

Comment:
Unless you're intentionally calling the callback once for every window, you should return here.